From b94cdbf8f792838a5f2db4d13c09048c6403992f Mon Sep 17 00:00:00 2001
From: Mark Salyzyn <salyzyn@google.com>
Date: Mon, 8 Dec 2014 15:06:52 -0800
Subject: [PATCH 516/634] pstore: remove superfluous memory size check

All previous checks will fail with error if memory size
is not sufficient to register a zone, so this legacy
check has become redundant.

Signed-off-by: Mark Salyzyn <salyzyn@android.com>
Acked-by: Kees Cook <keescook@chromium.org>
Change-Id: Ie21c988ae0b1ebb0dafa6c0c0b069e9cfe1e8506
---
 fs/pstore/ram.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/fs/pstore/ram.c b/fs/pstore/ram.c
index 75bdeaa..c637dd3 100644
--- a/fs/pstore/ram.c
+++ b/fs/pstore/ram.c
@@ -471,14 +471,6 @@ static int ramoops_probe(struct platform_device *pdev)
 	if (err)
 		goto fail_init_fprz;
 
-	if (!cxt->przs && !cxt->cprz && !cxt->fprz) {
-		pr_err("memory size too small, minimum is %zu\n",
-			cxt->console_size + cxt->record_size +
-			cxt->ftrace_size);
-		err = -EINVAL;
-		goto fail_cnt;
-	}
-
 	cxt->pstore.data = cxt;
 	/*
 	 * Console can handle any buffer size, so prefer LOG_LINE_MAX. If we
@@ -523,7 +515,6 @@ fail_buf:
 fail_clear:
 	cxt->pstore.bufsize = 0;
 	cxt->max_dump_cnt = 0;
-fail_cnt:
 	kfree(cxt->fprz);
 fail_init_fprz:
 	kfree(cxt->cprz);
-- 
1.9.1

